pkg/alertmanager/operator.go

```diff
@@ -933,6 +947,43 @@ receivers:
 	return nil
 }
 
+// find and delete global AlertmanagerConfig from amConfigs to avoid generating configuration twice
+func findAndDeleteGlobalAlertmanagerConfig(
+	am *monitoringv1.Alertmanager,
+	amConfigs map[string]*monitoringv1alpha1.AlertmanagerConfig,
+) (*monitoringv1alpha1.AlertmanagerConfig, error) {
+	crKey := types.NamespacedName{
+		Namespace: am.Namespace,
+		Name:      am.Spec.GlobalAlertmanagerConfig.Name,
+	}
+	amConfig, ok := amConfigs[crKey.String()]
```

Comment:
The global AlertmanagerConfig object might not be present in the `amConfigs` map. The operator should get it from the API.